// mpackage/getDetailedInfo.js
const generateRandomString = () => {
  const characters = "abcdefghijklmnopqrstuvwxyz0123456789";
  let randomString = "";

  for (let i = 0; i < 10; i++) {
    randomString += characters.charAt(
      Math.floor(Math.random() * characters.length)
    );
  }

  return randomString;
};

const calcAge = (birthDate) => {
  const currentDate = new Date();
  [day, month, year] = birthDate.split(".");
  let age = currentDate.getFullYear() - year;

  if (
    currentDate.getMonth() + 1 < month ||
    (currentDate.getMonth() + 1 == month && currentDate.getDate() < day)
  ) {
    age--;
  }

  return age;
};

const strFormat = (str) => {
  return str.charAt(0).toUpperCase() + str.toLowerCase().slice(1);
};

const getDetailedInfo = ({ name, dateBirth, purpose }) => {
  const detailedInfo = {
    id: generateRandomString(),
    firstName: strFormat(name.split(" ")[0]),
    lastName: strFormat(name.split(" ")[1]),
    dateBirth: dateBirth,
    age: calcAge(dateBirth),
    purpose: strFormat(purpose),
  };

  return detailedInfo;
};

module.exports = getDetailedInfo;
